Update glslang submodule

oltolm opened the pull request:
This is also necessary to fix the GCC build.

oltolm committed:
3rdparty: update glslang submodule

hrydgard commented:
Can you explain why we need to add two submodules? Also, this didn't quite build.

oltolm commented:
I think they are needed by glslang now, I am not sure, I'll double check. It does not build for different reasons. On Windows I didn't add the new submodules to MSVC solution because I don't use it. On other systems that use CMake the problems are different.

hrydgard commented:
Well, I don't want to add code that we're not using. I'm doubtful this is needed since it's not needed on the other platforms. We carry all the headers and stuff we need in the current dependencies.

oltolm commented:
This is from `glslang\CHANGES.md`
```
## 14.1.0 2024-03-08
* Add SPIRV-Tools-opt dependency if ENABLE_OPT
```

If I disable `ENABLE_OPT` then CMake prints the warning
```
spirv-tools not linked - illegal SPIRV may be generated for HLSL
```

To disable this warning `ENABLE_HLSL` needs to be disabled.
```
option(ENABLE_HLSL "Enables HLSL input support" ON)
```

What should I do? Should I disable `ENABLE_OPT` and `ENABLE_HLSL`?

hrydgard commented:
ENABLE_HLSL  should be disabled. We do not use HLSL as input to glslang currently.

hrydgard commented:
This PR still adds the submodules, please remove them.

oltolm commented:
I have removed them.

hrydgard commented:
Still doesn't build. Seems there's some inconsistency in C++ edition parameters?

oltolm commented:
I need some help here.

iOS build is failing with
```
/Applications/Xcode_15.4.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS17.5.sdk/usr/include/c++/v1/__filesystem/path.h:509:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
In file included from <built-in>:1:
In file included from /Users/runner/work/ppsspp/ppsspp/build-ios/ext/glslang/glslang/CMakeFiles/glslang.dir/cmake_pch.hxx:5:
In file included from /Users/runner/work/ppsspp/ppsspp/ext/glslang/glslang/MachineIndependent/pch.h:44:
In file included from /Users/runner/work/ppsspp/ppsspp/ext/glslang/glslang/MachineIndependent/SymbolTable.h:70:
/Users/runner/work/ppsspp/ppsspp/ext/glslang/glslang/MachineIndependent/../Include/InfoSink.h:112:60: error: 'string' is unavailable: introduced in iOS 13.0
                append(std::filesystem::absolute(location).string());
                                                           ^
```
Needs a newer iOS version?

macOS fails with
```
In file included from <built-in>:1:
In file included from /Users/runner/work/ppsspp/ppsspp/build/ext/glslang/glslang/CMakeFiles/glslang.dir/cmake_pch_x86_64.hxx:5:
In file included from /Users/runner/work/ppsspp/ppsspp/ext/glslang/glslang/MachineIndependent/pch.h:44:
In file included from /Users/runner/work/ppsspp/ppsspp/ext/glslang/glslang/MachineIndependent/SymbolTable.h:70:
/Users/runner/work/ppsspp/ppsspp/ext/glslang/glslang/MachineIndependent/../Include/InfoSink.h:112:50: error: 'path' is unavailable: introduced in macOS 10.15
                append(std::filesystem::absolute(location).string());
                                                 ^
```
needs a newer macOS version?

Android fails with
```
In file included from ../ext/glslang-build/../glslang/glslang/GenericCodeGen/Link.cpp:40:
In file included from ../ext/glslang-build/../glslang/glslang/GenericCodeGen/../Include/ShHandle.h:48:
../ext/glslang-build/../glslang/glslang/GenericCodeGen/../Include/InfoSink.h:108:20: error: no member named 'filesystem' in namespace 'std'; did you mean 'std::__fs::filesystem'?
            append(std::filesystem::absolute(shaderFileName).string());
                   ^~~~~~~~~~~~~~~
                   std::__fs::filesystem
/opt/hostedtoolcache/ndk/r21e/x64/sources/cxx-stl/llvm-libc++/include/filesystem:259:1: note: 'std::__fs::filesystem' declared here
_LIBCPP_BEGIN_NAMESPACE_FILESYSTEM
^
/opt/hostedtoolcache/ndk/r21e/x64/sources/cxx-stl/llvm-libc++/include/__config:763:58: note: expanded from macro '_LIBCPP_BEGIN_NAMESPACE_FILESYSTEM'
  _LIBCPP_BEGIN_NAMESPACE_STD namespace __fs { namespace filesystem {
                                                         ^
In file included from ../ext/glslang-build/../glslang/glslang/GenericCodeGen/Link.cpp:40:
In file included from ../ext/glslang-build/../glslang/glslang/GenericCodeGen/../Include/ShHandle.h:48:
../ext/glslang-build/../glslang/glslang/GenericCodeGen/../Include/InfoSink.h:112:24: error: no member named 'filesystem' in namespace 'std'; did you mean 'std::__fs::filesystem'?
                append(std::filesystem::absolute(location).string());
```
no idea why.

hrydgard commented:
Once #20000  is in, try rebasing on master.

hrydgard commented:
Hm, the Mac/iOS issue seems worse :/

hrydgard commented:
Hm, apparently we need to link to something too in the Android.mk / Locals.mk script...

I hate it when dependencies start using way too new C++ features just because..
